R/AzureDataLake.R

```diff
@@ -0,0 +1,368 @@
+    "https://", azureDataLakeAccount, ".azuredatalakestore.net/webhdfs/v1/",
+    relativePath,
+    "?op=CREATE&write=true",
+    "&api-version=2016-11-01"
+  )
+  if (!missing(overwrite)  && !is.null(overwrite)) URL <- paste0(URL, "&overwrite=", overwrite)
+  if (!missing(permission)  && !is.null(permission)) URL <- paste0(URL, "&permission=", permission)
+  
+  resHttp <- callAzureDataLakeApi(URL, verb = "PUT",
+                                  azureActiveContext = azureActiveContext,
+                                  content = contents, contenttype = "text/plain; charset=UTF-8",
+                                  verbose = verbose)
+  stopWithAzureError(resHttp)
+  # TODO: Check why this returns NULL
+}
+
+#' Azure Data Lake APPEND for specified relativePath of an azure data lake account.
+#'
+#' @inheritParams createAzureContext
+#' @inheritParams azureAuthenticate
+#'
+#' @param azureActiveContext Provide an `azureActiveContext` object used for authentication.
+#' @param azureDataLakeAccount Provide the name of the Azure Data Lake account.
+#' @param relativePath Provide a relative path of the directory.
+#' @param contents Provide contents to write to `relativePath`
+#' @param verbose Print tracing information (default FALSE).
+#'
+#' @return NULL
+#'
+#' @template
+#' @references
+#' @family Azure Data Lake Store functions
+#' @export
+azureDataLakeAppend <- function(azureActiveContext, azureDataLakeAccount, relativePath, contents = "", verbose = FALSE) {
+  
+  if (!missing(azureActiveContext) && !is.null(azureActiveContext)) {
+    assert_that(is.azureActiveContext(azureActiveContext))
+    azureCheckToken(azureActiveContext)
+  }
+  assert_that(is_storage_account(azureDataLakeAccount))
+  # TODO: Need a check for contents ?
+  #assert_that(is_content(contents))
+  verbosity <- set_verbosity(verbose)
+  
+  URL <- paste0(
+    "https://", azureDataLakeAccount, ".azuredatalakestore.net/webhdfs/v1/",
+    relativePath,
+    "?op=APPEND&append=true",
+    "&api-version=2016-11-01"
+  )
+  
+  resHttp <- callAzureDataLakeApi(URL, verb = "POST",
+                                  azureActiveContext = azureActiveContext,
+                                  content = contents, contenttype = "text/plain; charset=UTF-8",
+                                  verbose = verbose)
+  stopWithAzureError(resHttp)
+  # TODO: Check why this returns NULL
+}
+
+#' Azure Data Lake OPEN for specified relativePath of an azure data lake account.
+#'
+#' @inheritParams createAzureContext
+#' @inheritParams azureAuthenticate
+#'
+#' @param azureActiveContext Provide an `azureActiveContext` object used for authentication.
+#' @param azureDataLakeAccount Provide the name of the Azure Data Lake account.
+#' @param relativePath Provide a relative path of the directory.
+#' @param offset Provide the offset to read from.
+#' @param length Provide length of data to read.
+#' @param verbose Print tracing information (default FALSE).
+#'
+#' @return Returns a data frame.
```

Comment:
What's in it? The data read, obviously, but in what format?

Reply:
Fixed in https://github.com/omkarksa/AzureSMR/pull/1 will raise a pull request once this is merged into master